### ISSUE TITLE
Curried functions

### ISSUE THREAD

**Issue by danielo515** [phadej/typify#38]
Hello,
I didn't find any reference on the readme,so this is why I'm asking. Does this library support curried functions? If so, what are the limitations?
Regards

**Comment by phadej** [comment]
the auto-curried ones? I.e. may take 1,2,3... arguments and return either the final value or another function?

No, that's no supported, and probably won't be.

**Comment by danielo515** [comment]
No, I don't mean auto-curried (if you mean magic-partial-application a-la ramda). I mean actually curried functions, functions that only take a function and return another function 

```js
sum = a => b => a + b
```

Those ones, where you always know what the next return type will be. Funnily the function signature and the function definition will match 1:1
